comp110-worksheet-6/OxoBoard.cs

```diff
@@ -7,55 +7,270 @@
 namespace comp110_worksheet_6
 {
 	public enum Mark { None, O, X };
-
+	 
 	public class OxoBoard
 	{
+
+		//Initialise the board variable, which will be assigned to in OxoBoard().
+		Mark[,] board;
+		int numForARow;
+
 		// Constructor. Perform any necessary data initialisation here.
 		// Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
-		public OxoBoard(/* int width = 3, int height = 3, int inARow = 3 */)
+		public OxoBoard(int width = 3, int height = 3, int inARow = 3)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+			//Assigning values to variables
+			board = new Mark[width, height];
+			numForARow = inARow;
 		}
 
 		// Return the contents of the specified square.
 		public Mark GetSquare(int x, int y)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+			//Will return Mark for the specific board cell
+			return board[x, y];
 		}
 
 		// If the specified square is currently empty, fill it with mark and return true.
 		// If the square is not empty, leave it as-is and return False.
 		public bool SetSquare(int x, int y, Mark mark)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+			//Try and catch are used to stop values that are more than or less than the size of the grid.
+			try
+			{
+				if (board[x, y] == Mark.None)
+				{
+					//Assigns the board cell to the given mark paramater and returns true.
+					board[x, y] = mark;
+					return true;
+				}
+				else
+				{
+					//Return false if the square is already occupied.
+					return false;
+				}
+			}
+			catch
```

Comment:
It's best to `catch` specific exceptions rather than catching everything -- in this case you'd want to catch `IndexOutOfRangeException`